From 7e3f8f4f86d1fbd9ab1745fc0b1d00a19fb3c656 Mon Sep 17 00:00:00 2001
From: Daniel Kahn Gillmor <dkg@fifthhorseman.net>
Date: Wed, 2 Feb 2022 02:49:45 -0500
Subject: [PATCH 1/2] Drop windows and wasm targets

We can't build or test the windows targets, and we don't have all of
the dependencies for the wasm (web assembly) targets.

Dropping them from Cargo.toml leaves us building the expected
nettle-backed version of sequoia, which is appropriate for a debian
install.
Index: sequoia-openpgp/Cargo.toml
===================================================================
--- sequoia-openpgp.orig/Cargo.toml
+++ sequoia-openpgp/Cargo.toml
@@ -55,3 +55,0 @@ required-features = ["compression-deflat 1
-[dependencies.aes]
-version = "0.8"
-optional = true
@@ -65,7 +65,0 @@ required-features = ["compression-deflat 2
-[dependencies.block-padding]
-version = "0.3"
-optional = true
-
-[dependencies.blowfish]
-version = "0.9"
-optional = true
@@ -85,3 +69,0 @@ 3
-[dependencies.cast5]
-version = "0.11"
-optional = true
@@ -90,12 +90,0 @@ 4
-[dependencies.cipher]
-version = "0.4"
-features = ["std"]
-optional = true
-
-[dependencies.des]
-version = "0.8"
-optional = true
-
-[dependencies.digest]
-version = "0.10"
-optional = true
@@ -110,3 +110,0 @@ 5
-[dependencies.eax]
-version = "0.5"
-optional = true
@@ -105,7 +115,0 @@ 6
-[dependencies.ecdsa]
-version = "0.16"
-features = [
-    "hazmat",
-    "arithmetic",
-]
-optional = true
@@ -105,8 +120,0 @@ 7
-[dependencies.ed25519-dalek]
-version = "1"
-features = [
-    "rand",
-    "u64_backend",
-]
-optional = true
-default-features = false
@@ -134,7 +125,0 @@ 8
-[dependencies.generic-array]
-version = "0.14.4"
-optional = true
-
-[dependencies.idea]
-version = "0.5"
-optional = true
@@ -154,4 +130,0 @@ 9
-[dependencies.md-5]
-version = "0.10"
-features = ["oid"]
-optional = true
@@ -166,4 +140,0 @@ 10
-[dependencies.num-bigint-dig]
-version = "0.8"
-optional = true
-default-features = false
@@ -175,7 +150,0 @@ 11
-[dependencies.p256]
-version = "0.13"
-features = [
-    "ecdh",
-    "ecdsa",
-]
-optional = true
@@ -185,8 +155,0 @@ 12
-[dependencies.rand07]
-version = "0.7.3"
-optional = true
-package = "rand"
-
-[dependencies.rand_core]
-version = "0.6"
-optional = true
@@ -195,13 +160,0 @@ 13
-[dependencies.ripemd]
-version = "0.1"
-features = ["oid"]
-optional = true
-
-[dependencies.rsa]
-version = "0.9.0"
-optional = true
-
-[dependencies.sha-1]
-version = "0.10"
-features = ["oid"]
-optional = true
@@ -210,4 +170,0 @@ 14
-[dependencies.sha2]
-version = "0.10"
-features = ["oid"]
-optional = true
@@ -220,11 +180,0 @@ 15
-[dependencies.twofish]
-version = "0.7"
-optional = true
-
-[dependencies.typenum]
-version = "1.12.0"
-optional = true
-
-[dependencies.x25519-dalek-ng]
-version = "1"
-optional = true
@@ -276,8 +190,0 @@ 16
-crypto-cng = [
-    "cipher",
-    "eax",
-    "winapi",
-    "win-crypto-ng",
-    "ed25519-dalek",
-    "num-bigint-dig",
-]
@@ -290,31 +200,0 @@ 17
-crypto-rust = [
-    "aes",
-    "block-padding",
-    "blowfish",
-    "cast5",
-    "cfb-mode",
-    "cipher",
-    "des",
-    "digest",
-    "eax",
-    "ecb",
-    "ed25519",
-    "ed25519-dalek",
-    "generic-array",
-    "idea",
-    "md-5",
-    "num-bigint-dig",
-    "rand",
-    "rand07",
-    "ripemd",
-    "rsa",
-    "sha-1",
-    "sha2",
-    "twofish",
-    "typenum",
-    "x25519-dalek-ng",
-    "p256",
-    "rand_core",
-    "rand_core/getrandom",
-    "ecdsa",
-]
@@ -320,30 +210,0 @@
-[target."cfg(all(target_arch = \"wasm32\", target_os = \"unknown\"))".dependencies.chrono]
-version = "0.4.10"
-features = [
-    "std",
-    "wasmbind",
-]
-default-features = false
-
-[target."cfg(all(target_arch = \"wasm32\", target_os = \"unknown\"))".dependencies.getrandom]
-version = "0.2"
-features = ["js"]
-
-[target."cfg(all(target_arch = \"wasm32\", target_os = \"unknown\"))".dependencies.rand07]
-version = "0.7"
-features = ["wasm-bindgen"]
-package = "rand"
-
-[target."cfg(windows)".dependencies.win-crypto-ng]
-version = ">=0.4, <0.6"
-features = [
-    "rand",
-    "block-cipher",
-]
-optional = true
-
-[target."cfg(windows)".dependencies.winapi]
-version = "0.3.8"
-features = ["bcrypt"]
-optional = true
-default-features = false
